index.html

```diff
@@ -0,0 +1,110 @@
+<!-- Copyright 2024 [TODO: COPYRIGHT HOLDER NAME]
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. -->
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>UrlFilterParser</title>
+  <style>
+    body {
+      font-family: Arial, sans-serif;
+      margin: 20px;
+    }
+    input {
+      margin-bottom: 10px;
+      padding: 5px;
+      font-size: 16px;
+    }
+    pre {
+      background: #f4f4f4;
+      padding: 10px;
+      border: 1px solid #ddd;
+      width: 1000px;
+    }
+    button {
+      margin-top: 10px;
+      padding: 10px 20px;
+      border: none;
+      background-color: #007bff;
+      color: #fff;
+      border-radius: 5px;
+      cursor: pointer;
+    }
+    button:hover {
+      background-color: #0056b3;
+    }
+    .fileInfo {
+      margin-top: 20px;
+    }
+  </style>
+</head>
+<body>
+
+  <h1>URL Filter Parser</h1>
+  <input type="text" id="urlFilterInput" placeholder="Enter URL filter" size="50">
+  <button id="UrlFilterParseDemoButton">Parse URL Filter</button><br />
+  <input type="url" id="urlTestInput" placeholder="Enter test URL" size="50">
+  <button id="UrlTestButton">Test URL</button>
+  <h2>Parsed Rule:</h2>
+  <pre id="parsedRuleOutput"></pre>
+  <h2>Test Output:</h2>
+  <pre id="urlTestOutput"></pre>
+
+  <hr>
+
+  <h1>Request Tester</h1>
+  <form id="RequestTestForm">
+    <label for="httpMethod">HTTP Method:</label>
+    <select id="httpMethod" name="httpMethod">
+        <option value="GET">GET</option>
+        <option value="POST">POST</option>
+        <option value="PUT">PUT</option>
+        <option value="DELETE">DELETE</option>
+    </select><br />
+    <label for="httpRequestUrl">Request URL:</label>
+    <input type="url" id="httpRequestUrl" name="httpRequestUrl" placeholder="Request URL" required> <br />
+    <label for="httpHeaders">Headers (JSON format):</label>
+    <textarea id="httpHeaders" name="httpHeaders" placeholder='{"Content-Type": "application/json"}'></textarea><br />
+    <label for="httpBody">Body (for POST/PUT):</label>
+    <textarea id="httpBody" name="httpBody"></textarea><br />
+    <button type="submit" id="RequestTestButton">Test Request</button>
+  </form>
+  <br />
+  <label for="RequestOutput"><i>Request Object: </i></label>
```

Comment:
I don't believe it means anything to put a `for` pointing to a `pre` tag, so we might want to remove this / switch it to just a heading rather than a label: https://html.spec.whatwg.org/multipage/forms.html#category-label